omnibus/files/private-chef-ctl-commands/chef-server-ctl

```diff
@@ -0,0 +1,328 @@
+#!/opt/opscode/embedded/bin/ruby
+
+require 'rubygems'
+# Because we're not installed as a gem ourselves,
+# we'll need to explicitly load the omnibus-ctl gem
+# so that we can see it
+# so that we can see it.
+gem 'omnibus-ctl'
+require "omnibus-ctl"
+
+module Omnibus
+
+  # We keep the base behavior of Omnibus::Ctl, but
+  # add our own hooks for before/after command run.
+  # In particular we hook the service calls.
+  class ChefServerCtl < Omnibus::Ctl
+    def command_post_hook(*args)
+      command = args.shift
+      method = to_method_name("#{command}_post_hook")
+      if respond_to?(method)
+        send(method, *args)
+      end
+    end
+
+    # If a command has a pre-hook defined we will run it.
+    # Otherwise, if it is a run-sv command and the service it refers to
+    # is an external service, we will show an error since we
+    # can't control external services from here.
+    def command_pre_hook(*args)
```

Comment:
Similarly, it seems like likes 30-34 should be the default implementation of pre_hook.  The implementation could then perhaps just look at the return value from `super`.